docs/demo_examples/plot_cahn_hilliard.py

```diff
@@ -0,0 +1,116 @@
+"""
+Operators: Cahn Hilliard
+========================
+
+This example is based on the example in the FiPy_ library.
+Please see their documentation for more information about the
+Cahn-Hilliard equation.
+
+The "Cahn-Hilliard" equation separates a field \\\\( \\\\phi \\\\)
+into 0 and 1 with smooth transitions.
+
+.. math::
+
+    \\frac{\partial \phi}{\partial t} = \\nabla \cdot D \\nabla \left( \\frac{\partial f}{\partial \phi} - \epsilon^2 \\nabla^2 \phi \\right)
+
+Where \\\\( f \\\\) is the energy function \\\\( f = ( a^2 / 2 )\\\\phi^2(1 - \\\\phi)^2 \\\\)
+which drives \\\\( \\\\phi \\\\) towards either 0 or 1, this competes with the term
+\\\\(\\\\epsilon^2 \\\\nabla^2 \\\\phi \\\\) which is a diffusion term that creates smooth changes in \\\\( \\\\phi \\\\).
+The equation can be factored:
+
+.. math::
+
+    \\frac{\partial \phi}{\partial t} = \\nabla \cdot D \\nabla \psi \\\\
+    \psi = \\frac{\partial^2 f}{\partial \phi^2} (\phi - \phi^{\\text{old}}) + \\frac{\partial f}{\partial \phi} - \epsilon^2 \\nabla^2 \phi
+
+Here we will need the derivatives of \\\\( f \\\\):
+
+.. math::
+
+    \\frac{\partial f}{\partial \phi} = (a^2/2)2\phi(1-\phi)(1-2\phi)
+    \\frac{\partial^2 f}{\partial \phi^2} = (a^2/2)2[1-6\phi(1-\phi)]
+
+The implementation below uses backwards Euler in time with an
+exponentially increasing time step. The initial \\\\( \\\\phi \\\\)
+is a normally distributed field with a standard deviation of 0.1 and
+mean of 0.5. The grid is 60x60 and takes a few seconds to solve ~130
+times. The results are seen below, and you can see the field separating
+as the time increases.
+
+.. _FiPy: https://github.com/usnistgov/fipy
+
+.. http://www.ctcms.nist.gov/fipy/examples/cahnHilliard/generated/examples.cahnHilliard.mesh2DCoupled.html
+
+"""
+from __future__ import print_function
+import discretize
+from pymatsolver import Solver
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def run(plotIt=True, n=60):
+
+    np.random.seed(5)
+
+    # Here we are going to rearrange the equations:
+
+    # (phi_ - phi)/dt = A*(d2fdphi2*(phi_ - phi) + dfdphi - L*phi_)
+    # (phi_ - phi)/dt = A*(d2fdphi2*phi_ - d2fdphi2*phi + dfdphi - L*phi_)
+    # (phi_ - phi)/dt = A*d2fdphi2*phi_ + A*( - d2fdphi2*phi + dfdphi - L*phi_)
+    # phi_ - phi = dt*A*d2fdphi2*phi_ + dt*A*(- d2fdphi2*phi + dfdphi - L*phi_)
+    # phi_ - dt*A*d2fdphi2 * phi_ =  dt*A*(- d2fdphi2*phi + dfdphi - L*phi_) + phi
+    # (I - dt*A*d2fdphi2) * phi_ =  dt*A*(- d2fdphi2*phi + dfdphi - L*phi_) + phi
+    # (I - dt*A*d2fdphi2) * phi_ =  dt*A*dfdphi - dt*A*d2fdphi2*phi - dt*A*L*phi_ + phi
+    # (dt*A*d2fdphi2 - I) * phi_ =  dt*A*d2fdphi2*phi + dt*A*L*phi_ - phi - dt*A*dfdphi
+    # (dt*A*d2fdphi2 - I - dt*A*L) * phi_ =  (dt*A*d2fdphi2 - I)*phi - dt*A*dfdphi
+
+    h = [(0.25, n)]
+    M = discretize.TensorMesh([h, h])
```

Comment:
I realize this is probably copy-over, but lets try and stick with `mesh` instead of `M` for mesh objects in the examples